### PR TITLE
Update Docs to API 6.0.0

### LINKED ISSUE #601: API 6.0 and retirement of API 3.X docs
With the release of API 6.0, we no longer have the capacity to support <s>or host</s> the old API 3.X docs. We only want to offer three versions, to reduce load on Crowdin and shorten docs build times. Our agreed solution is to bundle the 3.X docs into a _pdf_ or _html_ file and offer that. At that point we can retire the 3.X branch and focus on versions 4, 5 and 6.

So, we need to (1) tweak the SpongeDocs to make a new branch for 6.1.0, including updating the sponge.conf file etc, <s>and (2) sort out publishing the finalised 3.X docs. A new docs page for these links would be reasonable, "Legacy versions", or maybe the "Sponge Retirement Home for Bewildered APIs".</s>
Any help or suggestions would be appreciated. @Tzky @12AwsomeMan34 @RobertHerhold @ST-DDT @ZephireNZ @saladoc @Minecrell

### PR DESCRIPTION
**Pull request by Inscrutable**
Resolves #601 
Pages to update:
 - [x] gradle
 - [x] maven
 - [x] sponge.conf
 - [x] mixins
 - [x] arguments
 - [x] debugging

### CODE REVIEW

**Comment by Spongy** [comment]
Setting up PR reference, please wait...

**Comment by johnfg10** [comment]
https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/9f278a7/plugin/project/index.html 
version also needs updating to version 6.0.0 stable and version 7.0.0-SNAPSHOT snapshot

File: source/server/getting-started/configuration/sponge-conf.rst [hunk 1]
```diff
@@ -232,7 +243,7 @@ This config was generated using SpongeForge build 2022 (with Forge 2202), Sponge
     # # If you need help with the configuration or have any questions related to Sponge,
     # # join us at the IRC or drop by our forums and leave a post.
     #
-    # # IRC: #sponge @ irc.esper.net ( https://webchat.esper.net/?channel=sponge )
+    # # IRC: #sponge @ irc.esper.net ( http://webchat.esper.net/?channel=sponge )
```
Comment (ST-DDT) [inline source/server/getting-started/configuration/sponge-conf.rst]: This should point to https

Comment (kashike) [inline source/server/getting-started/configuration/sponge-conf.rst]: https://github.com/SpongePowered/SpongeCommon/compare/4647bb6a6f14...192dac87a651

Comment (Meronat) [inline source/server/getting-started/configuration/sponge-conf.rst]: Why don't we just have it link to the IRC/chat page on SpongeHome to keep them in the Sponge website ecosystem.

Comment (Inscrutable) [inline source/server/getting-started/configuration/sponge-conf.rst]: I want to get this PR done ASAP. If more changes are needed in SpongeCommon, please get it done, or throw the lever and the docs can catch up later.

Comment (Meronat) [inline source/server/getting-started/configuration/sponge-conf.rst]: I don't think any changes are needed in SpongeCommon for this PR to go through. I'm just suggesting this link should just be changed to Sponge's own chat(IRC) page: https://www.spongepowered.org/chat

**Pull request merged**